Gem/Code/Source/Components/NetworkWeaponsComponent.cpp

```diff
@@ -360,10 +363,34 @@ namespace MultiplayerSample
                 const AZ::Vector3& aimAngles = GetNetworkSimplePlayerCameraComponentController()->GetAimAngles();
                 const AZ::Quaternion aimRotation =
                     AZ::Quaternion::CreateRotationZ(aimAngles.GetZ()) * AZ::Quaternion::CreateRotationX(aimAngles.GetX());
-                // TODO: This should probably be a physx raycast out to some maxDistance
                 const AZ::Vector3 fwd = AZ::Vector3::CreateAxisY();
-                const AZ::Vector3 aimTarget = worldTm.GetTranslation() + aimRotation.TransformVector(fwd * 5.0f);
-                AZ::Vector3 aimSource = weaponInput->m_shotStartPosition;
+                AZ::Vector3 baseCameraOffset;
+                AZ::Interface<AZ::IConsole>::Get()->GetCvarValue("cl_cameraOffset", baseCameraOffset);
+                const AZ::Vector3 cameraOffset = aimRotation.TransformVector(baseCameraOffset);
+                
+                // Setup a default aim target
+                AZ::Vector3 aimTarget = worldTm.GetTranslation() + cameraOffset + aimRotation.TransformVector(fwd * sv_WeaponsMaxRange);
+
+                // Cast the ray in the physics system from the center of the camera forward
+                if (auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get())
+                {
+                    if (AzPhysics::SceneHandle sceneHandle = sceneInterface->GetSceneHandle(AzPhysics::DefaultPhysicsSceneName);
+                        sceneHandle != AzPhysics::InvalidSceneHandle)
+                    {
+                        AzPhysics::RayCastRequest physicsRayRequest;
+                        physicsRayRequest.m_start = worldTm.GetTranslation() + cameraOffset;
+                        physicsRayRequest.m_direction = aimRotation.TransformVector(fwd);
+                        physicsRayRequest.m_distance = sv_WeaponsMaxRange;
+                        physicsRayRequest.m_queryType = AzPhysics::SceneQuery::QueryType::StaticAndDynamic;
+
+                        AzPhysics::SceneQueryHits result = sceneInterface->QueryScene(sceneHandle, &physicsRayRequest);
+                        if (result)
+                        {
+                            // Set target to first found intersect, if any
+                            aimTarget = result.m_hits[0].m_position;
+                        }
```

Comment:
maybe I'm not reading this accurately, but there should be some logic in here to make sure we don't aimTarget something behind the player - m_start should begin at some point in front of the player that they can actually hit.  We can either hard code this for now with a fixed offset, or use a vector/plane intersection to determine a reasonable point mathematically based on the player position, and forward direction.  The vector would start at m_start in the camera direction and intersect a plane (using AZ::Plane::CastRay) that has an origin of worldTm.GetTranslation() + fwd * someoffset with a normal of fwd or -fwd depending (using AZ::Plane::CreateFromNormalAndPoint)

Reply:
That shouldn't be necessary here given this logic is in a ComponentController (auth only). There's no actual camera object here. Instead we're deriving the camera position based on the transform of the player in question and the camera offset. That serves as the start position and then the ray direction is just based on the aim angles of the player. Given that, the aim target could only be in front unless the camera was made to face the front of the player for some reason.

Reply:
There isn't anything stopping a player from hitting an object that is between the camera and the player (behind the player). This will also be very noticeable until we add code that moves the camera in front of any geometry between the camera and the player, and even when we have that code, if we do not use the vector/plane intersection as the start, we'll have to update this code to determine on the server where the camera is.
For example in this screenshot, rotating the camera slightly causes the teleporter to be between the camera and the player.  This will be even more noticeable when you have ceiling/floor collisions

![image](https://user-images.githubusercontent.com/26804013/188197416-9feaf39f-9524-449b-8bfc-2930be54108b.png)

Reply:
There is another fun case, where the shooter can shoot with their head not being visible and someone who gets shot by this player thinks - "but that player couldn't see me" 
![image](https://user-images.githubusercontent.com/26804013/188199276-51fccaf7-2837-4c4d-896c-7f0fbf3991ec.png)

Reply:
The first case is a good point, I'll add a dot check to verify that. The latter seems valid in a TPS. If there's nothing obstructing the shot, it should connect.

Reply:
A dot check will prevent a hit on an object between the camera and player, but the only way to let the player shoot what's in front of them in that scenario is to move the start of the raycast down till it is in front of the player, or keep raycasting till you get a hit in front of the player.

Reply:
Yep, PhysX will give us a list of hits in the prescribed distance so we just ignore the ones that fail the dot check and go with the first one that succeeds.

Reply:
You'll need to update your PhysXQuery to do this using `m_reportMultipleHits = true` but, in general, a single vector/plane intersection will be less expensive than returning every hit along a ray from start to finish.

Reply:
Ah good to know on the flag. I'll go with report multiple hits for now. If the perf proves to be an issue we can always change it.